### PR TITLE
fix(test): add color global values support in css

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Enhancement

#### What is expected?

Text component now accepts global values for color property: `inherit`, `initial`, etc.

I also fixed SelectNumber story category to be Data Entry.

### CODE REVIEW

**Review by chambo-e — COMMENTED** [review]
I don't understand why we would need this ? 🤔

**Comment by matthprost** [comment]
> I don't understand why we would need this ? 🤔

We had many use case where we needed those properties with text and not set specific color or a color base on parent so to to avoid making:

```tsx
const StyledText = styled(Text)`
   color: inherit;
`

const Component = () => <StyledText />
```

we can directly pass the prop to the component.

**Comment by chambo-e** [comment]
> > I don't understand why we would need this ? 🤔
> 
> We had many use case where we needed those properties with text and not set specific color or a color base on parent so to to avoid making:
> 
> ```tsx
> const StyledText = styled(Text)`
>    color: inherit;
> `
> 
> const Component = () => <StyledText />
> ```
> 
> we can directly pass the prop to the component.

So only inherit I assume ?

Can't we have color:inherit by default on Text and have the color props optional instead ?

**Comment by matthprost** [comment]
> 

Okay sounds another good solution :)

File: src/components/Text/index.tsx [hunk 1]
```diff
@@ -45,13 +46,13 @@ const generateStyles = ({
...
 
-  const themeColor = theme.colors[color]
+  const themeColor = theme.colors[color as Color]
   const text = `text${definedProminence}${
     disabled ? 'Disabled' : ''
   }` as keyof typeof themeColor
 
   return `
-    color: ${theme.colors[color][text]};
+    color: ${color === 'inherit' ? color : theme.colors[color][text]};
```
Comment (chambo-e) [inline src/components/Text/index.tsx:55]: We could have 
```suggestion
    color: ${color ? theme.colors[color][text] : 'inherit'};
```

And avoid everything else :)

Comment (matthprost) [reply src/components/Text/index.tsx:55]: @chambo-e If I do that types won't show that inherit is the default value for color

Comment (chambo-e) [reply src/components/Text/index.tsx:55]: We don't need to, that's already how css works :)

**Comment by baptisteArno** [comment]
Can we make the `color` prop optional?

**Review by baptisteArno — CHANGES_REQUESTED** [review]
Can we make the `color` prop optional?

**Comment by matthprost** [comment]
> Can we make the `color` prop optional?

That's gonna be a two step migration, we make color required to add neutral to all `Text` then we will remove required and remove all text with color `inherit`

**Comment by baptisteArno** [comment]
What a pain @matthprost

**Pull request closed without merging**